src/core_editor/line_buffer.rs

```diff
@@ -454,6 +454,92 @@ impl LineBuffer {
     pub fn is_cursor_at_last_line(&self) -> bool {
         !self.get_buffer()[self.offset()..].contains('\n')
     }
+
+    /// Finds index for the first occurrence of a char to the right of offset
+    pub fn find_char_right(&self, c: &char) -> Option<usize> {
+        if self.offset() + 1 > self.lines.len() {
+            return None;
+        }
+
+        let search_str = &self.lines[self.offset() + 1..self.lines.len()];
+        search_str.find(*c).map(|index| index + self.offset() + 1)
```

Comment:
The `+1` has to be replaced with the `char::len_utf8` for the character starting at offset. `LineBuffer::grapheme_right_index()` finds the next index of a grapheme (potentially wider than the codepoint)

```suggestion

        let search_str = &self.lines[self.grapheme_right_index()..];
        search_str.find(*c).map(|index| index + self.grapheme_right_index())
```

Reply:
The second `self.grapheme_right_index()` was missing and causing panics on `f<non-ascii>` moves